### ISSUE TITLE
error when using `max-nesting-depth` with `"ignore": ["at-rules-without-declaration-blocks"]`

### ISSUE THREAD

**Issue by deepfriedmind** [stylelint/stylelint#885]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where using `max-nesting-depth` with `"ignore": ["at-rules-without-declaration-blocks"]` causes an error.

> Which rule, if any, is this issue related to?

`max-nesting-depth`

> What CSS is needed to reproduce this issue?

Not sure if it matters. I don't get any info about what it fails on.

> What stylelint configuration is needed to reproduce this issue?

```json
"max-nesting-depth": [3, {"ignore": ["at-rules-without-declaration-blocks"]}]
```

> Which version of stylelint are you using?

`4.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS plugin via gulp-postcss

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

I'm using SCSS syntax.

> What did you expect to happen?

No error.

> What actually happened (e.g. what warnings or errors you are getting)?

I get the following error:

```shell
TypeError in plugin 'gulp-postcss'
Message:
    Cannot read property 'every' of undefined
```

It works fine if I just set the rule to `"max-nesting-depth": 3`

**Comment by jeddy3** [comment]
Confirmed. Thanks for the report.

After a bit of investigation it looks like we need to be sure that the `@rules` we're checking *have blocks*. I believe the rule is tripping up on the likes of `a { @mixincontent; }`.
